Lib/idlelib/hyperparser.py

```diff
@@ -39,7 +39,7 @@ def index2line(index):
             for context in editwin.num_context_lines:
                 startat = max(lno - context, 1)
                 startatindex = repr(startat) + ".0"
-                stopatindex = "%d.end" % lno
+                stopatindex = "%d.end" % lno if end_at_eol else "end-1c"
```

Comment:
Since you're changing this line, an f-string would work make this more readable.